### PR TITLE
Changed color and layout of multipleSelection tag

### PR DESCRIPTION
**Pull request by perzonas**
## Background
The tag in columns for multiple select data has the same background color as the table background on every other row making it disappear.

The current design for multiple select is a combined list put inside a single tag making it hard to follow which teams there are

## Solution

Changed the design so that it is a single tag for each entry
Changed background color to a color that is not the same as the row background
Made the same changes in both old and new table

image

### CODE REVIEW

File: frontend/beCompliant/src/components/choiceTag/ChoiceTag.tsx [hunk 1]
```diff
@@ -17,15 +17,11 @@ function hasColorKey(obj: object): boolean {
...
-const findChoice = (choices: Choice[], cellValue: string) => {
-  return choices.find((choice) => choice.name === cellValue);
-};
-
 const getChoiceColors = (
   choices: Choice[],
   value: string | number
 ): ChoiceColors | undefined => {
-  const choice = findChoice(choices, value.toString());
+  const choice = choices.find((choice) => choice.name === value.toString());
```
Comment (perzonas) [inline frontend/beCompliant/src/components/choiceTag/ChoiceTag.tsx:24]: was just a one row function so moved it in directly

File: frontend/beCompliant/src/components/MobileFilter.tsx [hunk 2]
```diff
@@ -12,6 +11,7 @@ import {
   Flex,
   Text,
 } from '@kvib/react';
+import { TableMetaData } from '../types/tableTypes';
```
Comment (perzonas) [inline frontend/beCompliant/src/components/MobileFilter.tsx:14]: Errors I brought with me from my last pr 😅

File: frontend/beCompliant/src/components/Table.tsx [hunk 3]
```diff
@@ -7,13 +7,12 @@ import {
...
   useReactTable,
 } from '@tanstack/react-table';
-import { Choice, Field } from '../hooks/datafetcher';
-import { RecordType } from '../pages/TablePage';
 import { QuestionRow } from './questionRow/QuestionRow';
 import { DataTable } from './table/DataTable';
 import { DataTableCell } from './table/DataTableCell';
 import { DataTableHeader } from './table/DataTableHeader';
 import { Question } from './table/Question';
+import { Choice, Field, RecordType } from '../types/tableTypes';
```
Comment (perzonas) [inline frontend/beCompliant/src/components/Table.tsx:15]: same here

File: frontend/beCompliant/src/components/choiceTag/ChoiceTag.tsx [hunk 4]
```diff
@@ -40,31 +36,32 @@ const getChoiceColors = (
...
+      <Flex flexWrap={'wrap'} gap={'1'}>
+        {cellValue
+          .sort((a, b) => a.length - b.length)
+          .map((cell, index) => {
+            const choiceColors = getChoiceColors(choices, cell);
+            return (
+              <Tag
+                width="fit-content"
+                key={index}
+                backgroundColor={choiceColors?.backgroundColor}
```
Comment (perzonas) [inline frontend/beCompliant/src/components/choiceTag/ChoiceTag.tsx:52]: simplification as it returns undefined if it cant find it which is the same as '' in this case

File: frontend/beCompliant/src/components/choiceTag/ChoiceTag.tsx [hunk 5]
```diff
@@ -40,31 +36,32 @@ const getChoiceColors = (
...
-              backgroundColor={choiceColors ? choiceColors.backgroundColor : ''}
-              color={choiceColors ? choiceColors.textColor : ''}
-            >
-              {cellValue[index]}
-            </Tag>
-          );
-        })}
+      <Flex flexWrap={'wrap'} gap={'1'}>
+        {cellValue
+          .sort((a, b) => a.length - b.length)
```
Comment (perzonas) [inline frontend/beCompliant/src/components/choiceTag/ChoiceTag.tsx:45]: Sorting on the length of the value inside the tag to allow for the shortest values to be on the same row without wrapping. This does not affect how the table sorting works!  
Example how it would work:

[it] [short team]
[very long team name]

instead of it always being this:
[it]
[short]
[long team name trolololol]

File: frontend/beCompliant/src/components/table/DataTable.tsx [hunk 6]
```diff
@@ -15,7 +15,7 @@ export function DataTable<TData>({ table, showSearch = true }: Props<TData>) {
         <DataTableSearch alignSelf={'flex-end'} table={table} mr={10} />
       )}
       <TableContainer>
-        <Table variant="striped" colorScheme="gray">
+        <Table variant="striped">
```
Comment (perzonas) [inline frontend/beCompliant/src/components/table/DataTable.tsx:18]: there by default

File: frontend/beCompliant/src/components/table/Question.tsx [hunk 7]
```diff
@@ -27,8 +27,22 @@ export const Question = ({ value, column }: QuestionProps) => {
   }
 
   switch (column.type) {
-    case 'multipleSelects':
-      return <Box>{value}</Box>;
+    case 'multipleSelects': {
+      const valueArray = value.split(',');
```
Comment (perzonas) [inline frontend/beCompliant/src/components/table/Question.tsx:31]: In the columnDefinitions for the table we are joining the array with ',' for sorting and searching to work as intended. Done in `Table.tsx` lines 29-31

Here we are splitting it back into the original array

File: frontend/beCompliant/src/components/Table.tsx (hunk 3 above)
Comment (KrisAesoey) [reply frontend/beCompliant/src/components/Table.tsx:15]: yeah discovered as well this in my own PR 😆

**Review by KrisAesoey — APPROVED** [review]
Very clean, and great comments!
